src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Options/OtlpExporterOptionsBuilder.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Exporter
+{
+    public abstract class OtlpExporterOptionsBuilder<TBuilder> : ExporterOptionsBuilder<OtlpExporterOptions, TBuilder>
+        where TBuilder : OtlpExporterOptionsBuilder<TBuilder>
+    {
+        internal const string EndpointEnvVarName = "OTEL_EXPORTER_OTLP_ENDPOINT";
+        internal const string HeadersEnvVarName = "OTEL_EXPORTER_OTLP_HEADERS";
+        internal const string TimeoutEnvVarName = "OTEL_EXPORTER_OTLP_TIMEOUT";
+        internal const string ProtocolEnvVarName = "OTEL_EXPORTER_OTLP_PROTOCOL";
```

Comment:
I think this is discussion separate from this PR, but it's been on my mind that the processing of these environment variables should be on the exporter class itself not the options class (or the OptionsBuilder in this case).

For example let's I have the following environment variables set:

`OTLP_EXPORTER_OTLP_ENDPOINT=https://EndpointA:4317`
`OTLP_EXPORTER_OTLP_METRICS_ENDPOINT=https://EndpointB:4317`

Then I do:
```csharp
var options = new OtlpExporterOptions();
options.Endpoint = "http://EndpointC:4317";

var exporter = new OtlpMetricsExporter(options);
```

In this case I think the exporter should get configured with `https://EndpointB:4317`. This assumes that configuration precedences is environment variables first - I'm not actually sure this has been settled by the spec...